build.gradle

```diff
@@ -212,6 +212,107 @@ task printProjects() {
+}
+
+// TODO depend 'dist' on 'generateDistLicense'
+tasks.register('generateDistLicense') {
+    dependsOn('cyclonedxBom') // Task from 'org.cyclonedx.bom' plugin
+    doLast {
+        // Inputs
+        def bomFile = file('build/reports/bom.json')
+        def bom = new JsonSlurper().parseText(bomFile.text)
+        def projectLicenseText = file('LICENSE').text
+
+        // Outputs
+        def distLicenseFile = file('tools/dist-license/LICENSE')
+        def licensesDir = file('tools/dist-license/licenses/java/')
+        if (licensesDir.exists()) {
+            licensesDir.eachFile { it.delete() }
+        } else {
+            licensesDir.mkdirs()
+        }
+
+        List<Map<String, String>> thirdPartyArtifacts = new ArrayList<Map<String, String>>()
+        // Parse BOM
+        bom.components.each { component ->
+            // Exclude project modules
+            if (!component.group.startsWith('org.apache.eventmesh')) {
+                component.licenses.each { artifactLicense ->
+                    if (artifactLicense.license != null) {
+                        Map<String, String> artifact = new HashMap<String, String>()
+                        artifact.put("name", component.name)
+                        artifact.put("version", component.version)
+                        if (artifactLicense.license.id != null) {
+                            artifact.put("license", artifactLicense.license.id)
+                            if (artifactLicense.license.text != null) {
+                                artifact.put("licenseContent", new String(artifactLicense.license.text.content.decodeBase64()))
+                            }
+                        } else {
+                            artifact.put("license", artifactLicense.license.name)
+                            artifact.put("licenseContent", artifactLicense.license.url)
+                        }
+                        thirdPartyArtifacts.add(artifact)
+                    }
+                }
+            }
+        }
+        thirdPartyArtifacts.sort { a, b ->
+            def nameComparison = a.name <=> b.name
+            if (nameComparison == 0) {
+                return a.version <=> b.version
+            } else {
+                return nameComparison
+            }
+        }
+
+        def distLicenseText = projectLicenseText + "\n=======================================================================\n" +
+                "This distribution contains the following third-party:\n\n"
+        thirdPartyArtifacts.each { artifact ->
+            // Write licenses
+            def artifactLicenseFilename = artifact.license.replaceAll("/", "-") + ".txt"
+            def artifactLicenseFile = new File(licensesDir, artifactLicenseFilename)
+            if (artifact.licenseContent != null) {
+                artifactLicenseFile.text = artifact.licenseContent
+            } else {
+                artifactLicenseFile.text = "No license content provided by the artifact."
```

Comment:
This only happens with the `The GNU General Public License, v2 with Universal FOSS Exception, v1.0` license of `mysql-connector-j`, and only this artifact uses that license. 

You may notice that `mysql-connector-j` only has a license name in the [Maven repo](https://mvnrepository.com/artifact/com.mysql/mysql-connector-j/8.0.32) and no license content/url is recorded. It seems that this artifact can't be distributed under Apache 2.0 according to the practice of other projects: https://github.com/apache/eventmesh/issues/4842.

However, `apache/skywalking-eyes/dependency` actions don't deny `mysql-connector-j` when I generate eventmesh-dashboard's licenses. It seems to be an oversight of this action.

The license file of the unknown license can be removed/updated after this gradle task is completed. However, if an exception is thrown, the license file for subsequent licenses will not be generated. So maybe using a logger to output warnings would be a good compromise.

Reply:
![image](https://github.com/apache/eventmesh/assets/41445332/32bdb2e3-c9b3-4d86-a2f0-49e777448b31)

It seems there's another one artifact running into this case, and subsequent artifacts with the same license, but with the url set correctly, overwrite this file.

Reply:
> The license file of the unknown license can be removed/updated after this gradle task is completed. However, if an exception is thrown, the license file for subsequent licenses will not be generated. So maybe using a logger to output warnings would be a good compromise.

If this is not the only license check in the build script, it sounds good to me.

Note however that Apache projects can only bundle artifacts from a predefined list of licenses (cf.  [ASF 3rd party license policy](https://www.apache.org/legal/resolved), so the gradle task should only automatically handle licenses if it recognizes them. Otherwise manual intervention is required.